File: OneDrive/Documents/Data_Science/challenges/30-days-of-code/GO/Day5_ Reverse_a_String.go
package main

import "fmt"

func reverseString(str string) string {
    runes := []rune(str)
    for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }
    return string(runes)
}

func main() {
    var str string
    fmt.Print("Enter a string: ")
    fmt.Scan(&str)

    reversed := reverseString(str)
    fmt.Printf("Reversed string: %s\n", reversed)
}

File: OneDrive/Documents/Data_Science/challenges/30-days-of-code/C/Day5_ Reverse_a_String.c
#include <stdio.h>
#include <string.h>

void reverse_string(char* str) {
    int len = strlen(str);
    for (int i = 0; i < len / 2; i++) {
        char temp = str[i];
        str[i] = str[len - 1 - i];
        str[len - 1 - i] = temp;
    }
}

int main() {
    char str[100];
    printf("Enter a string: ");
    scanf("%s", str);

    reverse_string(str);

    printf("Reversed string: %s\n", str);

    return 0;
}
